### PR TITLE
Updates to crab submission for heppy jobs

### PR DESCRIPTION
**Pull request by peruzzim**
Reorganization of the code used to submit heppy jobs on crab.

See the README for running instructions and documentation of the options.

Features activated by command line options:
- choose the cfg configuration file and additional files to include (e.g. json)
- pass custom heppy options in crab jobs (in addition to the custom options, the isCrab option is set to True and can be used to tune the cfg configuration)
- choose the behaviour of AAA in remote jobs: free AAA access via redirector, usage of local files in remote site, or force reading from EOS via AAA. The useAAA flag of datasets should be always set to False in the cfg file.
- unpack some of the files produced in the job to have them returned separately from the output compressed archive
- choose remote running sites (whitelist)
- configurable output path with production tags and software version

### CODE REVIEW

**Comment by peruzzim** [comment]
The --AAAconfig=local option is not fully operational, as crab will submit jobs to remote sites without selecting those that have the requested dataset. They have to be whitelisted manually.

File: CMGTools/TTHAnalysis/cfg/crab/heppy_crab_config_env.py [hunk 1]
```diff
@@ -2,47 +2,42 @@
...
-    NJOBS = 4
-    NEVENTS = 200
+print "Will send dataset", os.environ["DATASET"], "with", os.environ["NJOBS"], "jobs"
 
-print "Will send dataset", dataset, "with", NJOBS, "jobs"
-
-config.General.requestName = dataset + "_" + cmg_version # task name
-config.General.workArea = 'crab_' + production_label # crab dir name
+config.General.requestName = os.environ["DATASET"] + "_" + os.environ["CMG_VERSION"] # task name
+config.General.workArea = 'crab_' + os.environ["DATASET"] # crab dir name
```
Comment (brucala) [inline CMGTools/TTHAnalysis/cfg/crab/heppy_crab_config_env.py]: Ciao Marco, all this is pretty cool, thanks!

just one little comment on the line above. If one repeats one dataset in different production, the crab working directory will be the same, no?. Why not keep crab_production_label as before for a cleaner bookkeeping?

**Comment by mariadalfonso** [comment]
@peruzzim this PR has conflicts to resolve

**Comment by peruzzim** [comment]
Confict fixed.

**Pull request merged**